runtime/shared/src/main/scala/org/http4s/googleapis/runtime/auth/AwsCredentials.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2024 Christopher Davenport
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package googleapis.runtime.auth
+import cats.MonadThrow
+import cats.data.EitherT
+import cats.data.OptionT
+import cats.data.Validated.Invalid
+import cats.data.Validated.Valid
+import cats.effect.Concurrent
+import cats.effect.Temporal
+import cats.effect.std.Env
+import cats.syntax.all._
+import fs2.io.IOException
+import io.circe.Decoder
+import org.http4s.headers.Accept
+
+import circe.jsonOf
+import CredentialsFile.ExternalAccount.ExternalCredentialSource
+import client.Client
+object AwsCredentials extends AwsSubjectTokenProvider {
+
+  /** @param id
+    *   GCP project id
+    * @param retrieveSubjectToken
+    *   abstract platform specific actions to retrieve subject token for token exchange
+    * @param externalAccount
+    *   configuration for external account credentials generation
+    * @param scopes
+    *   scopes for authorization
+    * @return
+    *   authorization token with impersonation
+    */
+  def apply[F[_]: Env](
+      client: Client[F],
+      id: String,
+      impersonationURL: Uri,
+      aws: ExternalCredentialSource.Aws,
+      externalAccount: CredentialsFile.ExternalAccount,
+      scopes: Seq[String],
+  )(implicit F: Temporal[F]): F[GoogleCredentials[F]] =
+    for {
+      urls <- validateSource(aws)
+      (r_cred_verification_url, maybe_url, maybe_region_url, maybe_imdsv2_sess_t_url) = urls
+
+      region = lookupAwsRegion(client, maybe_region_url)
+      secCredsSource = lookupSecurityCredentials(client)(maybe_url)
```

Comment:
This should be called inside retrieveSubjectTokenForAWS because role may change  during application lifecycle and session token is ephemeral value to be refreshed.